gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/UpdateEntityRequestValidator.java

```diff
@@ -43,7 +43,7 @@ public void validate(
       // only 1 attribute source is supported
       AttributeMetadata metadata = attributeMetadataMap.get(attributeId);
       Preconditions.checkArgument(
-          metadata.getSourcesCount() == 1 && metadata.getSources(0).equals(AttributeSource.EDS),
+          metadata.getSourcesCount() > 0 && metadata.getSourcesList().contains(AttributeSource.EDS),
```

Comment:
Not sure why this existed - probably because it invalidates the materialized data in QS? Anyway, the comment no long applies on line 43 so please remove